client-js/app/elements/labrad-grapher.ts

```diff
@@ -1,26 +1,43 @@
 import * as datavault from '../scripts/datavault';
 import {Places} from '../scripts/places';
 
+type ListItem = {
+  id: string,
+  name: string,
+  url: string,
+  isDir: boolean,
+  isDataset: boolean,
+  isParent: boolean,
+  starred: boolean,
+  trashed: boolean
+};
+
 @component('labrad-grapher')
 export class LabradGrapher extends polymer.Base {
   @property({type: Array, notify: true})
   path: Array<string>;
 
-  @property({type: Array})
+  @property({type: Array, value: () => { return []; }})
   dirs: Array<{name: string; url: string; tags: Array<string>}>;
 
-  @property({type: Array})
+  @property({type: Array, value: () => { return []; }})
   datasets: Array<{name: string; url: string; tags: Array<string>}>;
 
-  @property({type: String, notify: true})
-  selectedId: string;
+  @property({type: Array, value: () => { return []; }})
+  listItems: Array<ListItem>;
+
+  @property({type: Array, value: () => { return []; }})
+  listItems_: Array<ListItem>;
```

Comment:
Agreed, this is very confusing. I was thinking `listItems` and `listItemsView`, but `filteredListItems` sounds a better choice.